app/controllers/scholarship_oportunities_controller.rb

```diff
@@ -0,0 +1,35 @@
+class ScholarshipOportunitiesController < ApplicationController
+  def new
+    @scholarship_oportunity = ScholarshipOportunity.new
+  end
+
+  def index
+    @scholarship_oportunities = ScholarshipOportunity.all
+  end
+
+  def show
+    @scholarship_oportunity = ScholarshipOportunity.find(params[:id])
+    @students = @scholarship_oportunity.students
+  end
+
+  def create
+    @scholarship_oportunity = ScholarshipOportunity.new(scholarship_oportunity_params)
+    @scholarship_oportunity.name = @scholarship_oportunity.name.parameterize(separator: '_')
+    if ScholarshipOportunity.exists?(name: @scholarship_oportunity.name)
```

Comment:
Consider using `unless` if you're leaving the if branch empty

Reply:
Done